Cython/StringIOTree.py

```diff
@@ -57,6 +57,11 @@ def __init__(self, stream=None):
         self.write = stream.write
         self.markers = []
 
+    def empty(self):
+        if any(x.empty() for x in self.prepended_children):
+            return False
+        return not self.stream.tell()
```

Comment:
I'd reverse this. It seems simpler to check if anything was written to the stream itself, rather than testing all of its children first.
Also, I think the child condition needed to be negated, so I changed it to `all()` below.
```suggestion
        if self.stream.tell():
            return False
        return all(child.empty() for child in self.prepended_children)
```
(For the loop variable name change, it provides more information right away when reading the line from left to right. Otherwise, I'd have to wait unto the very end to understand that we're traversing the children here.)